### ISSUE TITLE
[Autocomplete] Explore and align on the name of the autocomplete component 

### ISSUE THREAD

**Issue by sadiesaurus** [Shopify/polaris-react#2267]
## Issue summary

TL;DR Let's align with the Discovery Experience team about what autocomplete means and explore whether it's the best fit as a name for this component. 

I've been chatting with @evija about the autocomplete component exploration and there's some concern about staying aligned with the Discovery Experience team in terms of terminology. 

As a part of the [Predictive Search project](https://vault.shopify.io/projects/5627), a lot of research and exploration has been done in the past year around defining the word "autocomplete" (especially since that's what their project was initially called). Their findings show that "After looking closer at what the autocomplete means, we found that there’s a strong connection to queries - completing a query which then takes you to a refined search results page." (See more information in context here: https://vault.shopify.io/projects/5627#roadmap_comment_33872) 

From my understanding, the description of autocomplete here doesn't match with the component that we're redesigning. I'd like to explore this and try to align with Discovery Experience. It would be great to maintain consistency and to feel confident that the name for this component is suitable.

**Comment by danrosenthal** [comment]
Have you considered combo box? https://en.m.wikipedia.org/wiki/Combo_box

**Comment by dleroux** [comment]
I like ComboBox. Since people use it in various ways, ComboBox sort of speaks more to the pattern.

**Comment by sadiesaurus** [comment]
> A combo box is a GUI feature that combines a drop-down box, list box, and/or an editable text field, giving the user multiple ways to input or select the desired information. (https://www.computerhope.com/jargon/c/combo_box.htm) 

Hmm I like it too, seems to fit well.
